### DISCUSSION TITLE
"Repository has been renamed - skipping" does not create warning

### DISCUSSION THREAD

**Discussion by hraftery** [renovatebot/renovate#19159] [Suggest an Idea]
If a target repository in the `repositories` list gets renamed, all that appears in the Renovate log is:

`INFO: Repository has been renamed - skipping`

and the repo is quietly skipped. In my mind, not producing an error if Renovate stops working on a repo is a bug. Further, an easy-win improvement would be to follow the rename and carry on like normal. The log message can then stay an `INFO` but be something like `INFO: Repository has been renamed - using new name`. This is in keeping with the generally transparent handling of superseded repo names on GitHub. Having to update the repo name in two places seems unnecessarily cumbersome.

Is this worthy of a bug report and a feature request?

**Comment by rarkins** [comment]
Is this your use case?
- You are providing a list of repositories, instead of autodiscovering all?
- The "old" repository is on the list, but the new is not?

**Reply by hraftery** [reply]
Yes exactly, sorry I should have mentioned that. In total, dozens of repos across several orgs, selected out of hundreds repos on an centrally managed opt-in basis. Renovate then works mostly silently, conveniently out of sight to many. So errors are important, as is robustness.

**Reply by rarkins** [reply]
In this case I would consider that you have an invalid bot config (you have specified a repo name which no longer exists) and therefore a log warning is most appropriate, so that you can become aware of it and fix the config. I'm happy for that log message to be a made a warning right away.

There are a few edge cases we need to consider if we were to "follow" the repo to the new one. For example Renovate may no longer have access to the new repo, especially if it's been moved/renamed to a new org.

Another factor is that in the case of autodiscovered repos, the "new" repo might already be in the list.

**Reply by hraftery** [reply]
Makes sense. I figured that since GitHub docs suggest renames are "backwards compatible" (or some words to that effect) and functions like `git` operations on the repo, and navigating to the "old" URL all quietly "work", that the "follow" mechanism was pretty robust. But it could well be due to a lot of checks and balances behind the scenes.
